internal/graph/graph.go

```diff
@@ -422,29 +422,42 @@ func (g *RelationshipGraph) getRelationshipEdgesWithTargetRewrite(
 //
 //	CycleDetectionCheckResolver  <-----|
 //		CachedCheckResolver              |
-//			LocalChecker                   |
-//				CycleDetectionCheckResolver -|
+//			DispatchThrottlingCheckResolver |
+//				LocalChecker                   |
+//					CycleDetectionCheckResolver -|
 //
 // The returned CheckResolverCloser should be used to close all resolvers involved in the
 // composition after you are done with the CheckResolver.
 func NewLayeredCheckResolver(
 	localResolverOpts []LocalCheckerOption,
 	cacheEnabled bool,
+	throttlingEnabled bool,
 	cachedResolverOpts []CachedCheckResolverOpt,
+	dispatchThrottlingCheckResolverOpts []DispatchThrottlingCheckResolverOpt,
 ) (CheckResolver, CheckResolverCloser) {
 	cycleDetectionCheckResolver := NewCycleDetectionCheckResolver()
+	var cachedCheckResolver *CachedCheckResolver
+	var dispatchThrottlingCheckResolver *DispatchThrottlingCheckResolver
 	localCheckResolver := NewLocalChecker(localResolverOpts...)
 
 	cycleDetectionCheckResolver.SetDelegate(localCheckResolver)
+	localCheckResolver.SetDelegate(cycleDetectionCheckResolver)
 
-	var cachedCheckResolver *CachedCheckResolver
 	if cacheEnabled {
 		cachedCheckResolver = NewCachedCheckResolver(cachedResolverOpts...)
-		cycleDetectionCheckResolver.SetDelegate(cachedCheckResolver)
 		cachedCheckResolver.SetDelegate(localCheckResolver)
+		cycleDetectionCheckResolver.SetDelegate(cachedCheckResolver)
 	}
 
-	localCheckResolver.SetDelegate(cycleDetectionCheckResolver)
+	if throttlingEnabled {
+		dispatchThrottlingCheckResolver = NewDispatchThrottlingCheckResolver(dispatchThrottlingCheckResolverOpts...)
+		dispatchThrottlingCheckResolver.SetDelegate(localCheckResolver)
+		if cacheEnabled {
+			cachedCheckResolver.SetDelegate(dispatchThrottlingCheckResolver)
+		} else {
+			cycleDetectionCheckResolver.SetDelegate(dispatchThrottlingCheckResolver)
```

Comment:
Can we have unit test to assert this?  See https://github.com/openfga/openfga/blob/aa623fb456294029d57152375ebc786899a4e5b7/pkg/server/server_test.go#L1567 as samples

Reply:
I couldn't find any test for this method. I think this has to be handled in the seperate PR we discussed to combine check resolver construction logic